File: scripts/icons.mjs
import fs from 'node:fs';
const actions=JSON.parse(fs.readFileSync('catalog/actions.json','utf8'));
const groups={
 code:['#167268','<path d="m25 18-13 13 13 13m22-26 13 13-13 13m-7-29-8 32"/>'],
 navigation:['#38628f','<path d="M12 19h19l6 6h23v24H12Z"/>'],
 editor:['#5464a0','<path d="m22 46 4-13 23-23 9 9-23 23Zm6-12 9 9M18 53h37"/>'],
 data:['#93652e','<ellipse cx="36" cy="16" rx="22" ry="8"/><path d="M14 16v29c0 11 44 11 44 0V16M14 30c0 11 44 11 44 0"/>'],
 git:['#ae5745','<circle cx="22" cy="15" r="5"/><circle cx="50" cy="15" r="5"/><circle cx="22" cy="48" r="5"/><path d="M22 20v23m0-10h15q13 0 13-13"/>'],
 layout:['#6f5a92','<rect x="12" y="10" width="48" height="42" rx="4"/><path d="M27 10v42M27 33h33M46 10v23"/>'],
 terminal:['#436e78','<rect x="11" y="11" width="50" height="41" rx="4"/><path d="m20 22 10 10-10 10m16 0h15"/>'],
 debug:['#a35570','<rect x="24" y="18" width="24" height="29" rx="11"/><path d="m27 18-6-7m24 7 6-7M13 24h11m24 0h11M12 34h12m24 0h12M17 46l9-6m20 0 9 6M36 19v28"/>'],
 notebook:['#3d768a','<rect x="18" y="8" width="39" height="47" rx="4"/><path d="M26 8v47M12 19h10M12 31h10M12 43h10m22-24 8 7-8 7"/>'],
 workflow:['#a1642d','<rect x="10" y="23" width="14" height="16" rx="3"/><rect x="48" y="23" width="14" height="16" rx="3"/><path d="M24 31h24m-9-8 9 8-9 8"/>'],
 r:['#487c51','<path d="M22 51V12h18c22 0 22 24 0 24H22m17 0 17 15"/>'],
 app:['#7b6931','<rect x="11" y="10" width="50" height="42" rx="4"/><path d="M11 22h50m-31 6 14 9-14 9Z"/>'],
 deploy:['#925075','<path d="M36 44V10m-13 13 13-13 13 13M14 39v14h44V39"/>']
};
fs.mkdirSync('docs/icons',{recursive:true});const dir='streamdeck/org.positron-deck.shortcuts.sdPlugin/imgs';fs.mkdirSync(dir,{recursive:true});
const esc=s=>s.replaceAll('&','&amp;').replaceAll('<','&lt;').replaceAll('"','&quot;');
for(const [index,a] of actions.entries()){
 const [color,shape]=groups[a.group]||groups.code;
 for(const [kind,size] of [['key',72],['list',20]])for(const [scale,suffix] of [[1,''],[2,'@2x']]){
 const bg=kind==='key'?`<rect width="72" height="72" rx="12" fill="${color}"/><rect x="1" y="1" width="70" height="70" rx="11" fill="none" stroke="#ffffff30"/>`:'';
 const svg=`<svg xmlns="http://www.w3.org/2000/svg" width="${size*scale}" height="${size*scale}" viewBox="0 0 72 72"><title>${esc(a.name)}</title>${bg}<g fill="none" stroke="white" stroke-width="3" stroke-linecap="round" stroke-linejoin="round" transform="translate(0 0)">${shape}</g></svg>`;
 fs.writeFileSync(`${dir}/${a.id}-${kind}${suffix}.svg`,svg);
 if(kind==='key'&&scale===1)fs.writeFileSync(`docs/icons/${a.id}.svg`,svg);
 }
}
console.log(`Generated icons for ${actions.length} actions.`);

File: streamdeck/scripts/icons.py
from pathlib import Path
from PIL import Image,ImageDraw
root=Path(__file__).resolve().parent.parent/'org.positron-deck.shortcuts.sdPlugin/imgs'
root.mkdir(parents=True,exist_ok=True)
shapes={
'code':'<path d="M28 22 16 34 28 46M44 22 56 34 44 46M40 18 32 49"/>',
'git':'<path d="M24 23V43Q24 50 34 50H43M24 32H37Q46 32 46 23"/><circle cx="24" cy="18" r="5"/><circle cx="46" cy="18" r="5"/><circle cx="48" cy="50" r="5"/>',
'r':'<path d="M25 50V18H38Q53 18 53 29T38 39H25M37 39 51 50"/>',
'app':'<rect x="15" y="18" width="43" height="33" rx="5"/><path d="M16 28H57M23 23H25M32 23H34M35 34 44 40 35 46Z"/>',
'deploy':'<path d="M36 46V18M25 29 36 18 47 29M18 42V52H54V42"/>'}
colors={'code':'#13526a','git':'#424778','r':'#305c46','app':'#655027','deploy':'#683744'}
for name,shape in shapes.items():
 for kind,size in [('list',20),('key',72)]:
  for scale,suffix in [(1,''),(2,'@2x')]:
   background=f'<rect x="1" y="1" width="70" height="70" rx="12" fill="{colors[name]}"/>' if kind=='key' else ''
   svg=f'<svg xmlns="http://www.w3.org/2000/svg" width="{size*scale}" height="{size*scale}" viewBox="0 0 72 72">{background}<g fill="none" stroke="#fff" stroke-width="3.5" stroke-linecap="round" stroke-linejoin="round">{shape}</g></svg>'
   (root/f'{name}-{kind}{suffix}.svg').write_text(svg)
for scale,suffix in [(1,''),(2,'@2x')]:
 (root/f'category{suffix}.svg').write_text(f'<svg xmlns="http://www.w3.org/2000/svg" width="{28*scale}" height="{28*scale}" viewBox="0 0 72 72"><g fill="none" stroke="#fff" stroke-width="4" stroke-linecap="round">{shapes["code"]}</g></svg>')
 size=256*scale;im=Image.new('RGBA',(size,size),'#102332');d=ImageDraw.Draw(im)
 def box(v):return tuple(int(x*scale) for x in v)
 d.rounded_rectangle(box((25,45,231,211)),radius=20*scale,fill='#1e586f',outline='#8ae3ff',width=4*scale)
 for row in range(2):
  for col in range(4):
   x=47+col*44;y=77+row*49;d.rounded_rectangle(box((x,y,x+30,y+30)),radius=5*scale,fill='#e4f8ff')
 d.rounded_rectangle(box((69,178,186,186)),radius=4*scale,fill='#8ae3ff')
 im.save(root/f'plugin{suffix}.png')
